src/material/schematics/ng-update/upgrade-rules/hammer-gestures-v9/hammer-gestures-rule.ts

```diff
@@ -134,7 +134,51 @@ export class HammerGesturesRule extends MigrationRule<null> {
     const hasCustomGestureConfigSetup =
         this._hammerConfigTokenReferences.some(r => this._checkForCustomGestureConfigSetup(r));
 
-    if (this._usedInRuntime || this._usedInTemplate) {
+    /*
+      Possible scenarios and how the migration should change the project:
```

Comment:
The structure of this comment doesn't seem to match the code. From reading the code, it seems like 2 and 3 should be sub-bullets of 1 and not sibling bullets.

Reply:
This should not match the implementation structure, but rather explain the individual cases and how we should react to. It would be difficult to make it match the implementation nesting etc.

FWIW: I also tried making this in a flow chart.. but it was too difficult and verbose.

Reply:
I don't think that you quite understood my point here, but I'll take a look at your next PR for this to see if I can explain things more clearly, if it's still a concern.

Reply:
What I understood here is that you propose to just move the bullets around so they match the nesting and structure of the actual implementation.

Reply:
The code seems to have two big branches related to 1., (a) - `hasCustomGestureConfigSetup` is true and (b) - `hasCustomGestureConfigSetup` is false. That doesn't seem to be communicated in this set of comments.

I looked through to see if the same behavior was performed for 2., 3., and 4. but that doesn't quite seem to be the case. 

For instance, 3. seems to match with the check in line 170 (`this._usedInTemplate` is true), but in that case none of the bullets in 3. are performed.

Then rule 2. (checking `_usedInRuntime `) only seems to be applied when `hasCustomGestureConfigSetup` is false, but these comments don't seem to capture that.

Reply:
Re: `hasCustomGestureConfigSetup === true`. (a) is handled by bullet point (1), while if `hasCustomGestureConfigSetup` is false, one of the remaining scenarios matches (2, 3, or 4).

Line 170 is a sub-scenario of (1) where a event from the gesture config is used.

Bullet point (2) is handled by line 190
Bullet point (3) is handled by 194
Bullet point (4) is handled by 197

This should cover all control flows possible by the current logic in `postAnalysis`.

Reply:
Yes. Rule 2 does not only apply if `hasCustomGestureConfig` is false. This is because if there is a custom gesture config _only_ (1) matches. The first matching scenario matches. Maybe that was unclear and can be improved.